outlines/testing-outline.md

```diff
@@ -0,0 +1,56 @@
+# Testing
+
+1. Testing your Meteor application
+  1. Basic intro to testing concepts
+  2. Unit testing to ensure a module works in isolation
+    1. Using stubs / mocks / spies to isolate a module
+  3. Integration testing to ensure parts of the system works together
+  4. End-to-end/acceptance testing to ensure the system works in totality
+  5. Load/stress testing to ensure the application can handle real world load
+2. Challenges of testing a Meteor application
+  1. The client/server divide, ensuring data is available on the client
+  2. Reactivity + testing the system responds to changing data properly
+3. Generating test data
+  1. How the test database is created/cleared
+  2. Defining factories using factory package
+  3. Stubbing collections to create test data client side
+4. Unit testing in Meteor: package testing
+  1. Right now a package is the best system we have for isolating code and running tests
+  2. Using practicalmeteor:mocha to run package tests with mocha
```

Comment:
One awkward thing is that we are recommending a different strategy for testing than Meteor itself uses - are there unknown problems here? I guess on the other hand there is tons of example code for mocha outside of the Meteor world, so it's probably actually better.